internal/webhook/kueue/validation_test.go

```diff
@@ -0,0 +1,275 @@
+			expectMessage: "Kueue label validation passed for \"$Kind\" in namespace \"test-ns\"",
+			operation:     admissionv1.Create,
+		},
+		{
+			name: "Valid Kueue label with other extra labels",
+			nsLabels: map[string]string{
+				"kueue.openshift.io/managed": "true",
+			},
+			kueueState: operatorv1.Managed,
+			objLabels: map[string]string{
+				"kueue.x-k8s.io/queue-name": "queue",
+				"random.label.io/something": "yes",
+			},
+			expectAllow:   true,
+			expectMessage: "Kueue label validation passed for \"$Kind\" in namespace \"test-ns\"",
+			operation:     admissionv1.Create,
+		},
+		{
+			name: "Incorrect Kueue label key",
+			nsLabels: map[string]string{
+				"kueue.openshift.io/managed": "true",
+			},
+			kueueState: operatorv1.Managed,
+			objLabels: map[string]string{
+				"kueue.x-k8s.io/queue-naem": "queue",
+			},
+			expectAllow:   false,
+			expectMessage: "Kueue label validation failed: missing required label \"kueue.x-k8s.io/queue-name\"",
+			operation:     admissionv1.Create,
+		},
+		{
+			name:     "Queue label present but namespace not labeled",
+			nsLabels: map[string]string{
+				// No kueue.openshift.io/managed label
+			},
+			kueueState: operatorv1.Managed,
+			objLabels: map[string]string{
+				"kueue.x-k8s.io/queue-name": "queue",
+			},
+			expectAllow:   true,
+			expectMessage: "Namespace \"test-ns\" is not labeled for Kueue (\"kueue.openshift.io/managed\"), skipping Kueue label validation",
+			operation:     admissionv1.Create,
+		},
+		{
+			name: "Namespace is Kueue-enabled with legacy label",
+			nsLabels: map[string]string{
+				"kueue-managed": "true",
+			},
+			kueueState: operatorv1.Managed,
+			objLabels: map[string]string{
+				"kueue.x-k8s.io/queue-name": "queue",
+			},
+			expectAllow:   true,
+			expectMessage: "Kueue label validation passed for \"$Kind\" in namespace \"test-ns\"",
+			operation:     admissionv1.Create,
+		},
+		{
+			name: "Kueue unmanaged state treated as enabled - success",
+			nsLabels: map[string]string{
+				"kueue.openshift.io/managed": "true",
+			},
+			kueueState: operatorv1.Unmanaged,
+			objLabels: map[string]string{
+				"kueue.x-k8s.io/queue-name": "queue1",
+			},
+			expectAllow:   true,
+			expectMessage: "Kueue label validation passed for \"$Kind\" in namespace \"test-ns\"",
+			operation:     admissionv1.Create,
+		},
+	}
+
+	// Included NoteBook as representative GVK since the webhook logic is metadata-based
+	// and does not vary across resource types (GVKs). Other GVKs like RayJob, etc.,
+	// are not included here to avoid redundant tests. Add them only if GVK-specific logic
+	// is introduced in the future.
+	supportedGVKs := []schema.GroupVersionKind{
+		{Group: "kubeflow.org", Version: "v1", Kind: "NoteBook"},
+	}
+
+	for _, tc := range testCases {
+		for _, gvk := range supportedGVKs {
+			t.Run(tc.name+"_"+gvk.Kind, func(t *testing.T) {
+				t.Parallel()
+
+				cli := newFakeClientWithObjects(sch, tc.nsLabels, tc.kueueState).Build()
```

Comment:
as follow up, it would be nice to also setup some tests using testenv so we can actually validate the interaction